iroha/src/domain.rs

```diff
@@ -99,9 +103,9 @@ pub mod isi {
                         .filter(|asset_id| asset_id.definition_id == asset_definition_id)
                         .cloned()
                         .collect::<Vec<_>>();
-                    keys.iter().for_each(|asset_id| {
+                    for asset_id in &keys {
                         let _ = account.assets.remove(asset_id);
-                    });
+                    }
```

Comment:
No need to modify this, iterators are zero cost abstractions https://doc.rust-lang.org/book/ch13-04-performance.html